volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -33,8 +37,30 @@ def get_requirements(cls):
             ),
         ]
 
-    def _generator(self):
-        layer = self.context.layers[self.config["primary"]]
+    @staticmethod
+    def enumerate_mft_records(
+        context: interfaces.context.ContextInterface,
+        config: interfaces.configuration.HierarchicalDict,
+        config_path: str,
+        attr_callback,
+    ) -> interfaces.objects.ObjectInterface:
+        try:
+            primary = context.layers[config["primary"]]
+        except KeyError:
+            vollog.error(
```

Comment:
Yeah, I admit do look a bit excessive, but at least it's clear what's going on now and having it split onto at least two lines will make it clearer if something goes wrong.